text/0617-cloudfront-oac-l2.md

```diff
@@ -0,0 +1,877 @@
+    this.protocol = protocol;
+    this.behavior = behavior;
+  }
+}
+
+/**
+ * An Origin Access Control.
+ * @resource AWS::CloudFront::OriginAccessControl
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cloudfront-originaccesscontrol.html
+ */
+export class OriginAccessControl extends Resource implements IOriginAccessControl {
+  /**
+   * Imports an origin access control from its id and origin type.
+   */
+  public static fromOriginAccessControlAttributes(scope: Construct, id: string, attrs: OriginAccessControlAttributes): IOriginAccessControl {
+    class Import extends Resource implements IOriginAccessControl {
+      public readonly originAccessControlId = attrs.originAccessControlId;
+      public readonly originAccessControlOriginType = attrs.originAccessControlOriginType;
+    }
+    return new Import(scope, id);
+  }
+  /**
+   * Creates an origin access control for a S3 origin.
+   */
+  public static forS3(scope: Construct, id: string, props?: CommonOriginAccessControlProps): IOriginAccessControl {
+    return new OriginAccessControl(scope, id, {
+      ...props,
+      originAccessControlOriginType: OriginAccessControlOriginType.S3,
+    });
+  }
+
+  public readonly originAccessControlId: string;
+  public readonly originAccessControlOriginType: string;
+
+  constructor(scope: Construct, id: string, props: OriginAccessControlProps) {
+    super(scope, id);
+    this.originAccessControlOriginType = props.originAccessControlOriginType;
+
+    // Check if origin access control name is 64 characters or less
+    if (props.originAccessControlName) {
+      this.validateOriginAccessControlName(props.originAccessControlName);
+    }
+
+    const resource = new CfnOriginAccessControl(this, 'Resource', {
+      originAccessControlConfig: {
+        description: props.description,
+        name: props.originAccessControlName ?? Names.uniqueResourceName(this, {
+          maxLength: 64,
+        }),
+        signingBehavior: props.signing?.behavior ?? SigningBehavior.ALWAYS,
+        signingProtocol: props.signing?.protocol ?? SigningProtocol.SIGV4,
+        originAccessControlOriginType: this.originAccessControlOriginType,
+      },
+    });
+
+    this.originAccessControlId = resource.attrId;
+  }
+
+  private validateOriginAccessControlName(name: string) {
+    if (!Token.isUnresolved(name) && name.length > 64) {
+      throw new Error(`Origin access control name must be 64 characters or less, '${name}' has length ${name.length}`);
+    }
+  }
+}
+```
+
+#### Modifications to `S3BucketOrigin` class
+
+The `S3BucketOrigin` will have two methods, `withAccessIdentity()` and `withAccessControl()`, which each return a class configured with
+the corresponding method of origin access control.
+
+In the case where an imported bucket is being used for the S3 origin, calling `bucket.addToResourcePolicy()` will fail to add the policy statement. Existing
```

Comment:
We spend a lot of time in this RFC dedicated to this imported buckets issue. It's fine to call out the known risks of using an imported bucket in a concise bulleted list, but we don't need all this prose discussing it this extensively.